packages/build/src/plugins_core/build_command.js

```diff
@@ -14,10 +25,12 @@ const coreStep = async function ({
   nodePath,
   childEnv,
   logs,
+  constants: { PUBLISH_DIR },
   netlifyConfig: {
     build: { command: buildCommand, commandOrigin: buildCommandOrigin },
   },
 }) {
+  await cleanupBlobsDir(buildDir, PUBLISH_DIR)
```

Comment:
If I'm understanding this correctly, this effectively means we're saying that no steps before `onBuild` can create blobs since these will be removed, are we ok with that?

Reply:
Yeap. I wasn't quite sure of where to put it, to be honest. An `onPreBuild` or something that runs very very first would be ideal, but wasn't sure how to do that in a clean way. Any pointers appreciated!

Reply:
Ping @eduardoboucas here as well. Not sure how big of an issue it is (we can document it if it goes live like this maybe?).

Reply:
> Yeap. I wasn't quite sure of where to put it, to be honest. An onPreBuild or something that runs very very first would be ideal, but wasn't sure how to do that in a clean way. Any pointers appreciated!

I'm thinking probably the cleanest way is to have a `onPreBuild` core step that runs before everything else and where we can do any cleanup (this is one example, but we could potentially use it for anything else 🤷). Happy to leave that for a further PR.

Reply:
> I'm thinking probably the cleanest way is to have a `onPreBuild` core step that runs before everything else and where we can do any cleanup (this is one example, but we could potentially use it for anything else 🤷).

I agree!

Reply:
I have a draft PR for the changes: https://github.com/netlify/build/pull/5413
If OK, I can push those here, otherwise we can merge this one and keep iterating on that one.

Reply:
Another side-effect of the current approach is that we won't do the cleanup if your site doesn't have a build command, right? I know it's a bit of an edge case, but I think it means we could end up pushing blobs from a previous build?

That draft PR looks good to me. If you want to push those changes here, go for it!

Reply:
> That draft PR looks good to me. If you want to push those changes here, go for it!

Agree 👍 also happy to have it as a follow up, however y'all prefer.

> Another side-effect of the current approach is that we won't do the cleanup if your site doesn't have a build command, right?

Ah right!! Good point 👍

Reply:
> Another side-effect of the current approach is that we won't do the cleanup if your site doesn't have a build command, right?

Correct. Not ideal, agree.

I'll merge this to get things going and prep the other PR as a fast-follow.